packages/devtools_server/lib/src/server.dart

```diff
@@ -233,3 +259,17 @@ void printOutput(
 }) {
   print(machineMode ? jsonEncode(json) : message);
 }
+
+/// Map the URI (which may already be Observatory web app) to a WebSocket URI
+/// for the VM service. If the URI is already a VM Service WebSocket URI it
+/// will not be modified.
+Uri getVmServiceUriFromObservatoryUri(Uri uri) {
```

Comment:
Instead of duplicating this method from
package:devtools we should add this method to 
`package:vm_service_lib`
so we can use it everywhere.

Reply:
Makes sense - I've added a TODO, and will open a PR in `vm_service_lib` today.

Reply:
PR is at https://github.com/dart-lang/vm_service_drivers/pull/232